app/controllers/topics_controller.rb

```diff
@@ -840,9 +840,10 @@ def timings
   end
 
   def feed
+    raise Discourse::NotFound if Post.where(topic_id: params[:topic_id]).count == 0
```

Comment:
You can do much shorter using `exists?` 😉 

```suggestion
    raise Discourse::NotFound unless Post.exists?(topic_id: params[:topic_id])
```